src/megaclient.cpp

```diff
@@ -6667,17 +6667,16 @@ void MegaClient::mapuser(handle uh, const char* email)
         // yes: add email reference
         u = &users[hit->second];
 
-        if (strcmp(u->email.c_str(), email))
-        { 
-            if (u->email.size())
-            {
-                umindex.erase(u->email);
```

Comment:
I think that the new approach doesn't take into account email changes and wouldn't remove the element related to the previous email from `umindex`.